### PR TITLE
Eager conter initialization, to fix initial counts problem

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala [hunk 1]
```diff
@@ -75,14 +77,21 @@ class FlinkProcessCompiler(creator: ProcessConfigCreator,
     )
   }
 
+  //TODO: should this be configurable somehow?
+  private def prepareDefaultListeners(usedNodes: UsedNodes) = {
+    //see comment in Interpreter.interpretNext
+    //TODO: how should we handle branch?
+    val nodesHandledInNextParts = (nodeData: NodeData) => nodeData.isInstanceOf[CustomNode] || nodeData.isInstanceOf[node.Sink]
```
Comment (gadomsky) [inline engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala:84]: shouldn't be `EndingCustomNode` also included here?

File: engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/FlinkProcessRegistrar.scala [hunk 2]
```diff
@@ -91,11 +93,12 @@ class FlinkProcessRegistrar(compileProcess: (EspProcess, ProcessVersion, Deploym
 
     def nodeContext(nodeComponentId: NodeComponentInfo, validationContext: Either[ValidationContext, Map[String, ValidationContext]]): FlinkCustomNodeContext = {
       val exceptionHandlerPreparer = (runtimeContext: RuntimeContext) =>
-        compiledProcessWithDeps(runtimeContext.getUserCodeClassLoader).prepareExceptionHandler(runtimeContext)
+        //FIXME: None??
```
Comment (arkadius) [inline engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/FlinkProcessRegistrar.scala:96]: How to fix it? :)

File: engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala [hunk 3]
```diff
@@ -102,3 +111,11 @@ class FlinkProcessCompiler(creator: ProcessConfigCreator,
     }
   }
 }
+
+//The logic of listeners invocation is a bit tricky for CustomNodes/Sinks (see Interpreter.interpretNode)
+//Metric listeners have to initialize counters in appropriate parts...
+private[process] case class UsedNodes(nodes: Iterable[NodeData], nextParts: Iterable[String])
```
Comment (arkadius) [inline engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala:117]: existence of `nodes` are pretty obvious here, but I don't get why `nextParts` are also necessary. can you write a few words about that in comment?

Comment (mproch) [reply engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala:119]: I added longer comment...

File: engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/FlinkProcessRegistrar.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/FlinkProcessRegistrar.scala]: I removed it, it was left from some early experiments :)

**Review by arkadius — APPROVED** [review]

**Pull request merged**